perfkitbenchmarker/linux_benchmarks/iperf_benchmark.py

```diff
@@ -98,63 +128,241 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, thread_count,
-  # STDOUT: ------------------------------------------------------------
-  # Client connecting to 10.237.229.201, TCP port 5001
-  # TCP window size: 0.04 MByte (default)
-  # ------------------------------------------------------------
-  # [  6] local 10.76.234.115 port 53527 connected with 10.237.229.201 port 5001
-  # [  3] local 10.76.234.115 port 53524 connected with 10.237.229.201 port 5001
-  # [  4] local 10.76.234.115 port 53525 connected with 10.237.229.201 port 5001
-  # [  5] local 10.76.234.115 port 53526 connected with 10.237.229.201 port 5001
-  # [ ID] Interval       Transfer     Bandwidth
-  # [  4]  0.0-60.0 sec  3730 MBytes  521.1 Mbits/sec
-  # [  5]  0.0-60.0 sec  3499 MBytes   489 Mbits/sec
-  # [  6]  0.0-60.0 sec  3044 MBytes   425 Mbits/sec
-  # [  3]  0.0-60.0 sec  3738 MBytes   522 Mbits/sec
-  # [SUM]  0.0-60.0 sec  14010 MBytes  1957 Mbits/sec
-
-  thread_values = re.findall(r'\[SUM].*\s+(\d+\.?\d*).Mbits/sec', stdout)
-  if not thread_values:
-    # If there is no sum you have try and figure out an estimate
-    # which happens when threads start at different times.  The code
-    # below will tend to overestimate a bit.
-    thread_values = re.findall(r'\[.*\d+\].*\s+(\d+\.?\d*).Mbits/sec', stdout)
-
-    if len(thread_values) != thread_count:
-      raise ValueError('Only %s out of %s iperf threads reported a'
-                       ' throughput value.' %
-                       (len(thread_values), thread_count))
-
-  total_throughput = 0.0
-  for value in thread_values:
-    total_throughput += float(value)
-
-  metadata = {
-      # The meta data defining the environment
-      'receiving_machine_type': receiving_vm.machine_type,
-      'receiving_zone': receiving_vm.zone,
-      'sending_machine_type': sending_vm.machine_type,
-      'sending_thread_count': thread_count,
-      'sending_zone': sending_vm.zone,
-      'runtime_in_seconds': FLAGS.iperf_runtime_in_seconds,
-      'ip_type': ip_type
-  }
-  return sample.Sample('Throughput', total_throughput, 'Mbits/sec', metadata)
+
+  if protocol == TCP:
+
+    iperf_cmd = ('iperf -e --client %s --port %s --format m --time %s -P %s' %
+                 (receiving_ip_address, IPERF_PORT,
+                  FLAGS.iperf_runtime_in_seconds,
+                  thread_count))
+    if FLAGS.iperf_tcp_per_stream_bandwidth:
+      iperf_cmd = iperf_cmd + (' -b %dM' % FLAGS.iperf_tcp_per_stream_bandwidth)
+    # the additional time on top of the iperf runtime is to account for the
+    # time it takes for the iperf process to start and exit
+    timeout_buffer = FLAGS.iperf_timeout or 30 + thread_count
+    stdout, _ = sending_vm.RemoteCommand(iperf_cmd, should_log=True,
+                                         timeout=FLAGS.iperf_runtime_in_seconds + timeout_buffer)
+
+    multi_thread = re.findall(r'\[SUM\]\s+\d+\.\d+-\d+\.\d+\s\w+\s+\d+\s\w+\s+\d+\s\w+\/\w+\s+\d+\/\d+\s+\d+\s+', stdout)
+    window_size = re.findall(r'TCP window size: \d+\.\d+ \S+', stdout)
```

Comment:
Since you use these values later on I would just get them here:

window_size_match = re.search(r'TCP window size: (?P<size>\d+\.\d+) (?P<units>\S+)', stdout) 

if window_size_match:
   window_size_num = float(window_size_match.group('size'))
   window_size_measurement = window_size.match.group('units')